### ISSUE TITLE
image download suggestion

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#433]
I have just been rummaging (as I does) and have noticed that on some pages images are being downloaded from the internet. I was expecting these images to already be in place

I realise that part of the problem is that not all the namespaces are used to build html pages (and hence determining any images to download)

This got me thinking, rather than selecting which namespaces to use, use the information that comes from the \<format> field within the original xml dump of the pages.

The thinking goes something like this

Either add an extra column to the 'page' table (page_format_id) or create a new table (id, format_id) that only contains non text/x-wiki entries (as there are many more wikitext pages than non wikitext)

Armed with this info, use it to generate the list of html pages to be built. (only wikitext pages)

This formatting information could also be useful for implementing syntax highlighting (among other things)

What do you think?

**Comment by gnosygnu** [comment]
> Either add an extra column to the 'page' table (page_format_id) or create a new table (id, format_id) that only contains non text/x-wiki entries (as there are many more wikitext pages than non wikitext)

I think this is a solid idea. I've been wanting to add this for a while, as I need a better way to differentiate pages in the MediaWiki namespace.

> This formatting information could also be useful for implementing syntax highlighting (among other things)

Syntax highlighting is rough though. Last I looked, MediaWiki uses a Python library for syntax highlighting which I balked at trying to port over.

> I realise that part of the problem is that not all the namespaces are used to build html pages (and hence determining any images to download)

I saved this for last, since it's more of a curiosity thing? Which pages / namespaces did not have images? Offhand, I would think it's images in the Template and File namespace....

**Comment by desb42** [comment]
> Which pages / namespaces did not have images? 

Depends on the wiki
But Modules (/doc), Help, Wikipedia, Portal
In wikisource, Author and Index (Page, with some other code changes)

Hence by keeping track of the page format it becomes less wiki dependent

**Comment by gnosygnu** [comment]
Agreed. This is the better way to go. My only concern is that I've never parsed all the pages in Help / Templates / Modules (doc) before, so I don't know what may break.
